src/python/pants/engine/internals/graph.py

```diff
@@ -174,20 +174,90 @@ async def find_owners(owners_request: OwnersRequest) -> Owners:
 
     # Walk up the buildroot looking for targets that would conceivably claim changed sources.
     candidate_specs = tuple(AscendantAddresses(directory=d) for d in dirs_set)
-    candidate_targets = await Get(Targets, AddressSpecs(candidate_specs))
+    candidate_explicit_targets = await Get(Targets, AddressSpecs(candidate_specs))
```

Comment:
What would the effect be of selecting candidate targets using `FilesystemSpecs` here? Could the generated sub-target logic live inside `Specs` expansion instead?